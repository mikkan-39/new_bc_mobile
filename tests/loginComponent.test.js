import * as React from 'react';
import LoginComponent from '../src/components/loginComponent';
import { myTestRenderer } from './myTestRenderer';
import { render, screen, fireEvent } from '@testing-library/react-native';


describe('loginComponent test', () => {
	it('should match snapshot', () => {
		const tree = myTestRenderer(<LoginComponent />)
		expect(tree).toMatchSnapshot();
	});

	it('should call callback on button press', () => {
		// TODO:
	});
});