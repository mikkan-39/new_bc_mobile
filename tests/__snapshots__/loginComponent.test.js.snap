// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginComponent test should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#eee",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "flex-start",
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "aspectRatio": 1,
          "backgroundColor": "#fff",
          "borderRadius": 30,
          "flexShrink": 2,
          "height": undefined,
          "marginBottom": "10%",
          "marginTop": "30%",
          "maxWidth": 300,
          "padding": 10,
          "width": "50%",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/logo.png",
          }
        }
        style={
          {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
    <TextInput
      onChangeText={[Function]}
      style={
        {
          "backgroundColor": "#fff",
          "borderRadius": 15,
          "color": "#000",
          "fontSize": 20,
          "height": 50,
          "marginTop": 25,
          "maxWidth": 300,
          "padding": 10,
          "paddingHorizontal": 15,
          "width": "75%",
        }
      }
      testID="username"
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      secureTextEntry={true}
      style={
        {
          "backgroundColor": "#fff",
          "borderRadius": 15,
          "color": "#000",
          "fontSize": 20,
          "height": 50,
          "marginTop": 25,
          "maxWidth": 300,
          "padding": 10,
          "paddingHorizontal": 15,
          "width": "75%",
        }
      }
      testID="password"
      value=""
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#976dd7",
          "borderRadius": 15,
          "height": 50,
          "justifyContent": "center",
          "margin": 25,
          "maxWidth": 200,
          "opacity": 1,
          "width": "50%",
        }
      }
      testID="loginButton"
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 24,
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
